spec/controllers/followings_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'spec_helper'
+
+describe FollowingsController do
+  before { request.env['HTTP_REFERER'] = "from_whence_I_came" }
+
+  describe "GET index" do
+    let(:alice) { Fabricate(:user) }
+    let(:billy) { Fabricate(:user) }
+    let(:following) { Fabricate(:following, user: alice, followee: billy) }
+    let(:action) { get :index }
+
+    it "assigns @followings to current user's followings" do
+      set_current_user(alice)
+      action
```

Comment:
I am not sure if refactoring of this type is that helpful. Think about it that one has to look what action means here,